### PR TITLE
DROOLS-4317: [DMN Designer] Documentation - Add the "Download .doc file" button to the documentation tab

### PR DESCRIPTION
**Pull request by danielzhe**
As discussed in e-mail, this option was changed for now to "Download HTML", since the approach that we are using to generate .doc have some issues with MS Word.

### CODE REVIEW

**Comment by manstis** [comment]
@danielzhe Is this dependent on the other docs related PR?

https://github.com/kiegroup/kie-wb-common/pull/2739

**Comment by danielzhe** [comment]
@manstis No.

File: kie-wb-common-dmn/kie-wb-common-dmn-client/src/main/java/org/kie/workbench/common/dmn/client/editors/documentation/common/HTMLDownloadHelper.java [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    private final static String FILE_EXTENSION = ".HTML";
+    private final static String ENCODING = "data:text/html;charset=utf-8,";
+
+    public void download(final String filename, final String html) {
+
+        final String sourceHTML = HEADER + html + FOOTER;
+        final String source = Global.encodeURIComponent(sourceHTML);
+
+        final HTMLAnchorElement fileDownload = (HTMLAnchorElement) document.createElement("a");
+        document.body.appendChild(fileDownload);
```
Comment (manstis) [inline kie-wb-common-dmn/kie-wb-common-dmn-client/src/main/java/org/kie/workbench/common/dmn/client/editors/documentation/common/HTMLDownloadHelper.java]: Sweet!

File: kie-wb-common-dmn/kie-wb-common-dmn-client/src/main/java/org/kie/workbench/common/dmn/client/editors/documentation/DMNDocumentationView.java [hunk 2]
```diff
@@ -82,6 +94,16 @@ public void onPrintButtonClick(final ClickEvent e) {
         printHelper.print(documentationContent);
     }
 
+    @EventHandler("download-html-file")
+    public void onDownloadHtmlFile(final ClickEvent e) {
+        final String html = getCurrentDocumentationHTML();
+        downloadHelper.download(DOCUMENTATION_FILENAME, html);
```
Comment (jomarko) [inline kie-wb-common-dmn/kie-wb-common-dmn-client/src/main/java/org/kie/workbench/common/dmn/client/editors/documentation/DMNDocumentationView.java]: Just idea, could we suffix the `DOCUMENTATION_FILENAME` with real file name to distinguish multiple downloads? Or is that computation too much expensive?

```java
private String getCurrentDocumentationFilename() {
        return getDiagram()
                .map(diagram -> documentationService.processDocumentation(diagram))
                .map(dmnDocumentation -> DOCUMENTATION_FILENAME + "-" + dmnDocumentation.getFileName())
                .orElse(DOCUMENTATION_FILENAME);
    }
```

**Comment by danielzhe** [comment]
@jomarko Changes applied!

**Pull request merged**